rts/motoko-rts/src/idl.rs

```diff
@@ -55,8 +56,20 @@ const IDL_PRIM_lowest: i32 = -17;
 const IDL_EXT_blob: i32 = -129;
 const IDL_EXT_tuple: i32 = -130;
 
+unsafe fn leb128_decode_32(buf: *mut Buf) -> u32 {
+    let value = leb128_decode(buf);
+    assert!(value <= u32::MAX as usize);
+    value as u32
+}
+
+unsafe fn sleb128_decode_32(buf: *mut Buf) -> i32 {
```

Comment:
There'd be much fewer diffs below if you just renamed these decoders without the _32 suffix (but beware recursion in the bodies)

Reply:
Thanks for the suggestion. I refactored this.